src/javaRosa.js

```diff
@@ -309,20 +314,36 @@ define([
             return this.updateForMug(mug, mug.getLabelValue());
         },
         updateForMug: function (mug, defaultLabelValue) {
+            function getPresence(itext) {
+                if (_.isFunction(itext.presence)) {
+                    return itext.presence(mug.options);
+                }
+                return itext.presence;
+            }
```

Comment:
I think this should be a method on Mug (`mug.getPresence("labelItext")`), so you wouldn't have to re-implement the protocol here and every other place where it's needed. That can be done on a separate PR though.